### PR TITLE
Add statementId and sessionId to rawlog table

### PR DESCRIPTION
**Pull request by gavinyue**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #10351

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/cnservice/server_task.go [hunk 1]
```diff
@@ -122,6 +122,24 @@ func (s *service) createSQLLogger(command *logservicepb.CreateTaskService) {
...
+		nil,
+		nil)
+	pu.StorageEngine = s.storeEngine
+	pu.TxnClient = s._txnClient
+	s.cfg.Frontend.SetDefaultValues()
+	pu.FileService = s.fileService
+	pu.LockService = s.lockService
+	moServerCtx := context.WithValue(context.Background(), config.ParameterUnitKey, pu)
+
+	ieFactory := func() ie.InternalExecutor {
```
Comment (xzxiong) [inline pkg/cnservice/server_task.go:138]: 如果CN服务已经ready，提供另一个选择：
```
v, ok := runtime.ProcessLevelRuntime().GetGlobalVariables(runtime.InternalSQLExecutor)
if !ok {
	return moerr.New...
}
executor := v.(executor.SQLExecutor)
```

File: pkg/util/export/table/table.go [hunk 2]
```diff
@@ -303,6 +305,33 @@ type TableOptions interface {
...
 
+func (tbl *Table) ToUpgradeSql(ctx context.Context) string {
+	if tbl.UpgradeColumns == nil || len(tbl.UpgradeColumns) == 0 {
+		return ""
+	}
+
+	var columnDefinitions []string
+
+	// Generate ADD COLUMN statements for each upgrade column
+	for _, actions := range tbl.UpgradeColumns {
```
Comment (xzxiong) [inline pkg/util/export/table/table.go:316]: 这里版本号 没有实现相关逻辑，需要重点备注。// TODO 或 // fixme

File: pkg/cnservice/server_heartbeat.go [hunk 3]
```diff
@@ -90,6 +90,7 @@ func (s *service) handleCommands(cmds []logservicepb.ScheduleCommand) {
 		if cmd.CreateTaskService != nil {
 			s.createTaskService(cmd.CreateTaskService)
 			s.createSQLLogger(cmd.CreateTaskService)
+			s.upgradeSystemTable()
```
Comment (xzxiong) [inline pkg/cnservice/server_heartbeat.go:93]: 执行的机制有问题，这样每个CN 每次重启都会执行 ？
这个不合理。

File: pkg/util/trace/impl/motrace/schema.go [hunk 4]
```diff
@@ -129,13 +129,24 @@ var (
...
+	sessionIDCol    = table.SpanIDStringColumn("session_id", "session id")
+	statementIDCol  = table.SpanIDStringColumn("statement_id", "statement id")
 	spanKindCol     = table.StringColumn("span_kind", "span kind, enum: internal, statement, remote")
 	parentSpanIDCol = table.SpanIDStringColumn("parent_span_id", "parent span uniq id")
 	spanNameCol     = table.StringColumn("span_name", "span name, for example: step name of execution plan, function name in code, ...")
 	startTimeCol    = table.DatetimeColumn("start_time", "start time")
 	endTimeCol      = table.DatetimeColumn("end_time", "end time")
 	resourceCol     = table.TextDefaultColumn("resource", `{}`, "static resource information")
 
+	UpgradeColumns = map[string]map[string][]table.Column{
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/schema.go:141]: 变量名可以更明确些。例如, 最啰嗦的写法 rawlogTableUpdateColumns

File: pkg/cnservice/server_heartbeat.go (hunk 3 above)
Comment (gavinyue) [reply pkg/cnservice/server_heartbeat.go:93]: 因为Upgrade是可能会时常发生的， 如果不每次都执行，  就只能额外指定执行， 

比如 mo-service --upgrade xxxx

这里必须确保upgrade sql 是idempotent的。

**Comment by gavinyue** [comment]
Need this bug to be fixed to pass the compatibility bvt test

**Review by daviszhen — APPROVED** [review]

File: pkg/frontend/util.go [hunk 5]
```diff
@@ -503,16 +503,12 @@ func logInfo(ses *Session, info string, msg string, fields ...zap.Field) {
 		return
 	}
 	fields = append(fields, zap.String("session_info", info))
-	sessionId := ""
-	statementId := ""
 	if ses != nil {
-		sessionId = strconv.Itoa(int(ses.GetConnectionID()))
 		if ses.tStmt != nil {
-			statementId = string(ses.tStmt.StatementID[:])
+			fields = append(fields, zap.ByteString("session_id", ses.tStmt.SessionID[:]))
```
Comment (xzxiong) [inline pkg/frontend/util.go:508]: when ses.tStmt is nil, session_id still can get from `ses.GetUUID()`

Comment (xzxiong) [reply pkg/frontend/util.go:508]: the key should be const value

File: pkg/util/trace/impl/motrace/report_log.go [hunk 6]
```diff
@@ -103,6 +105,8 @@ func (m *MOZapLog) FillRow(ctx context.Context, row *table.Row) {
 	row.SetColumnVal(messageCol, table.StringField(m.Message))
 	row.SetColumnVal(extraCol, table.StringField(m.Extra))
 	row.SetColumnVal(stackCol, table.StringField(m.Stack))
+	row.SetColumnVal(statementIDCol, table.UuidField(m.StatementID[:]))
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/report_log.go:108]: should ignore `m.StatementID == [16]byte{}` case.
save `00000000-0000-0000-0000-000000000000` value is wast

File: pkg/util/trace/impl/motrace/report_log.go [hunk 7]
```diff
@@ -135,6 +139,14 @@ func ReportZap(jsonEncoder zapcore.Encoder, entry zapcore.Entry, fields []zapcor
 			}
 			continue
 		}
+
+		if v.Type == zapcore.ByteStringType {
+			if v.Key == "session_id" {
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/report_log.go:144]: the key better be const val

**Review by xzxiong — APPROVED** [review]

**Review by fengttt — COMMENTED** [review]
There need to be a doc somewhere that contains a sql query to retrieve all related errors/infos.

**Review by reusee — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Review by badboynt1 — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Pull request merged**